src/QorePGConnection.cpp

```diff
@@ -308,9 +308,11 @@ static AbstractQoreNode* qpg_data_timestamp(char *data, int type, int len, QoreP
    return DateTimeNode::makeAbsolute(conn->getTZ(), nv, us);
 }
 
+// the DATEOID format is a signed 32-bit integer giving the day offset from 2000-01-01
 static AbstractQoreNode* qpg_data_date(char *data, int type, int len, QorePGConnection *conn, const QoreEncoding *enc) {
-   int val = (ntohl(*((uint32_t *)data)) + 10957) * 86400;
-   return new DateTimeNode((int64)val);
+    int32_t val = ntohl(*((int32_t*)data));
+    int64 v = (static_cast<int64>(val) + 10957) * 86400;
```

Comment:
just fyi, I would like to not use c-style cast, and rather use c++ static/reinterpret casts eg https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting

also there is int64 mixing with int32_t, uint32_t, and the `int64_t` now also exists and it's a standard in c++, but since this is a bugfix for 0.8.13 I would say, let's merge it as-is, and change this approach for the future